Dashboard/helpers.py

```diff
@@ -0,0 +1,189 @@
+    '''Returns an ses client from a configuration.'''
+    return boto_client(
+        'ses',
+        aws_access_key_id = config["ACCESS_KEY"],
+        aws_secret_access_key = config['SECRET_ACCESS_KEY'],
+        region_name = config['AWS_REGION_NAME']
+    )
+
+def is_ses(boto_ses_client: ses_client) -> bool:
+    '''Returns true if ses client else false.'''
+    return (isinstance(boto_ses_client, BaseClient)
+            and boto_ses_client._service_model.service_name == 'ses')  # pylint: disable=protected-access
+
+def can_parse_as_float(to_check: any) -> bool:
+    """Determines if a value can be parsed as a float.
+    Returns True on success, else False."""
+    try:
+        float(to_check)
+        return True
+    except (ValueError, TypeError):
+        logging.error("Input can't be parsed as a float")
+        return False
+
+def get_supported_websites(conn: connection) -> list:
+    """Gets a list of supported websites."""
+    with get_cursor(conn) as cur:
+        cur.execute("SELECT website_name from websites;")
+        websites = cur.fetchall()
+    return [website["website_name"] for website in websites]
+
+def get_user_id(conn: connection, email: str) -> int | None:
+    """Get a user id from the database based on an email."""
+    with get_cursor(conn) as cur:
+        cur.execute("SELECT user_id from users WHERE email = %s;", (email,))
+        user_id = cur.fetchone()
+    if user_id == None:
+        return None
+    return user_id.get("user_id")
+
+def get_product_id(conn: connection, url: str) -> int | None:
+    """Gets the products id from the database given a url."""
+    with get_cursor(conn) as cur:
+        cur.execute("SELECT product_id from products WHERE url = %s;", (url,))
+        product_id = cur.fetchone()
+    if product_id == None:
+        return None
+    return product_id.get("product_id")
+
+def is_subscription_in_table(conn: connection, user_id: int, product_id: int) -> bool:
+    """Determines if a user is subscribed to a particular product."""
+    with get_cursor(conn) as cur:
+        cur.execute("""SELECT * FROM subscriptions
+                        WHERE user_id = %s AND product_id = %s""",
+                    (user_id, product_id))
+        counts = cur.fetchone()
+    return counts is not None
+
+def insert_subscription_into_db(
+        conn: connection, user_id: int, product_id: int,
+        price_threshold: float | int | None) -> bool:
+    """Inserts a new subscription into the subscriptions table.
+    Returns True on success, else False."""
+    if isinstance(price_threshold, (int, float)) and price_threshold <= 0:
+        return False
+    try:
+        with get_cursor(conn) as cur:
+            cur.execute("""INSERT INTO subscriptions
+                         (user_id, product_id, price_threshold)
+                        VALUES (%s, %s, %s);""", (user_id, product_id, price_threshold))
+            conn.commit()
+        return True
+    except Exception:
```

Comment:
Use a more specific error

Reply:
This is client facing, so preventing errors from reaching customers if necessary